Core/CRFSharp/encoder/ModelWriter.cs

```diff
@@ -5,24 +5,32 @@
 using System.Text;
 using AdvUtils;
 using System.Threading.Tasks;
+using CRFSharp.decoder;
 
 namespace CRFSharp
 {
-    public class ModelWritter : BaseModel
+    public class ModelWriter : BaseModel
     {
+        private readonly string modelFileName;
+
         private readonly Pool<StringBuilder> _buildersPool =
             new Pool<StringBuilder>(p => new StringBuilder(100), b => b.Clear());
 
+        private readonly ModelReader modelReader;
```

Comment:
If modelReader is used for incremental training, A new class named ModelReaderWriter or ModelStream would be a better name.